code/frameworks/sveltekit/src/mocks/app/state.svelte.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * Inspired by the the code:
+ * {@link https://github.com/sveltejs/kit/blob/main/packages/kit/src/runtime/client/state.svelte.js}
+ *
+ * The differences:
+ *
+ * - Legacy Svelte support is not included
+ * - Not using classes (internal coding style preference)
+ *
+ * @module
+ */
+import { fn } from 'storybook/test';
+
+/**
+ * @typedef {Object} App
+ * @property {Object} Error
+ * @property {string} Error.message
+ * @property {Object} Locals
+ * @property {Object} PageData
+ * @property {Object} PageState
+ * @property {Object} Platform
+ */
+
+/**
+ * @typedef {Object} Page
+ * @property {URL} url
+ * @property {Record<string, string>} params
+ * @property {Object} route
+ * @property {string | null} route.id
+ * @property {number} status
+ * @property {App.Error | null} error
+ * @property {App.PageData & Record<string, any>} data
+ * @property {App.PageState} state
+ * @property {any} form
+ */
+
+/** @type {Page['data']} */
+let pageData = $state.raw({});
+/** @type {Page['form']} */
+let pageForm = $state.raw(null);
+/** @type {Page['error']} */
+let pageError = $state.raw(null);
+/** @type {Page['params']} */
+let pageParams = $state.raw({});
+/** @type {Page['route']} */
+let pageRoute = $state.raw({ id: null });
+/** @type {Page['state']} */
+let pageState = $state.raw({});
+/** @type {Page['status']} */
+let pageStatus = $state.raw(-1);
+/** @type {Page['url']} */
+let pageUrl = $state.raw(new URL(location.origin));
+
+/** @type {Page} */
+export let page = {
+  data: pageData,
+  form: pageForm,
+  error: pageError,
+  params: pageParams,
+  route: pageRoute,
+  state: pageState,
+  status: pageStatus,
+  url: pageUrl,
+};
```

Comment:
If you want this to be updated it can't just have a reference to the raw state...all of this needs to be getters/setters
```suggestion
export let page = {
  get data() {
    return pageData;
  },
  set data(newPageData: typeof pageData) {
    pageData = newPageData;
  },
  get form() {
    return pageForm;
  },
  set form(newPageForm: typeof pageForm) {
    pageForm = newPageForm;
  },
  get error() {
    return pageError;
  },
  set error(newPageError: typeof pageError) {
    pageError = newPageError;
  },
  get params() {
    return pageParams;
  },
  set params(newPageParams: typeof pageParams) {
    pageParams = newPageParams;
  },
  get route() {
    return pageRoute;
  },
  set route(newPageRoute: typeof pageRoute) {
    pageRoute = newPageRoute;
  },
  get state() {
    return pageState;
  },
  set state(newPageState: typeof pageState) {
    pageState = newPageState;
  },
  get status() {
    return pageStatus;
  },
  set status(newPageStatus: typeof pageStatus) {
    pageStatus = newPageStatus;
  },
  get url() {
    return pageUrl;
  },
  set url(newPageUrl: typeof pageUrl) {
    pageUrl = newPageUrl;
  },
};
```